program/src/state/mod.rs

```diff
@@ -49,126 +47,87 @@ impl RewardCenter {
         &self,
         listing_price: u64,
         payout_operation: &PayoutOperation,
-    ) -> u64 {
+    ) -> Result<u64> {
         match payout_operation {
             PayoutOperation::Multiple => {
                 msg!("Payout operation mode: Multiple");
-                return listing_price
+                listing_price
                     .checked_mul(self.reward_rules.payout_numeral.into())
-                    .ok_or(ListingRewardsError::NumericalOverflowError)
-                    .unwrap();
+                    .ok_or(RewardCenterError::NumericalOverflowError.into())
             }
 
             PayoutOperation::Divide => {
                 msg!("Payout operation mode: Divide");
-                return listing_price
+                listing_price
                     .checked_div(self.reward_rules.payout_numeral.into())
-                    .ok_or(ListingRewardsError::NumericalOverflowError)
-                    .unwrap();
+                    .ok_or(RewardCenterError::NumericalOverflowError.into())
             }
         }
     }
 
-    // TODO: review the effects of decimals on the payouts. The math is clean when the currency token is the same as the reward token.
     pub fn payouts(&self, listing_price: u64) -> Result<(u64, u64)> {
         let total_token_payout = self
-            .calculate_total_token_payout(listing_price, &self.reward_rules.mathematical_operand);
+            .calculate_total_token_payout(listing_price, &self.reward_rules.mathematical_operand)?;
 
         let seller_share = self.reward_rules.seller_reward_payout_basis_points;
 
         let seller_payout = (seller_share as u128)
             .checked_mul(total_token_payout as u128)
             .and_then(|product| product.checked_div(10000))
-            .ok_or(ListingRewardsError::NumericalOverflowError)? as u64;
+            .ok_or(RewardCenterError::NumericalOverflowError)? as u64;
 
         let buyer_payout = total_token_payout
             .checked_sub(seller_payout)
-            .ok_or(ListingRewardsError::NumericalOverflowError)?;
+            .ok_or(RewardCenterError::NumericalOverflowError)?;
 
         Ok((seller_payout, buyer_payout))
     }
 }
 
 #[account]
 pub struct Listing {
-    pub is_initialized: bool,
     pub reward_center: Pubkey,
     pub seller: Pubkey,
     pub metadata: Pubkey,
     pub price: u64,
     pub token_size: u64,
     pub bump: u8,
     pub created_at: i64,
-    pub canceled_at: Option<i64>,
-    pub purchase_ticket: Option<Pubkey>,
 }
 
 impl Listing {
     pub fn size() -> usize {
         8 + // delimiter
-        1 + // is_initialized
         32 + // reward_center
         32 + // seller
         32 + // metadata
         8 + // price
         8 + // token_size
         1 + // bump
-        8 + // created_at
-        1 + 8 + // canceled_at
-        1 + 32 // purchase_ticket
```

Comment:
Removing canceled_at and purchase_ticket assignment to listing. Instead going to index instructions and update the accounts. Same for offers.